Multimodal Textbox (Chat Input Component) (#7420)
gradio-app/gradio: guides/04_chatbots/01_creating-a-chatbot-fast.md

).launch()
```

## Add Multimodal Capability to your chatbot

> You may want to add multimodal capability to your chatbot. For example, you may wantusers to be able to easily upload images or files to your chatbot and ask questions about it. This is supported by the `gr.MultimodalTextbox()` component, which supports both text and file inputs. Here's an example of how to use it:

Reviewer: ```suggestion
You may want to add multimodal capability to your chatbot. For example, you may want users to be able to easily upload images or files to your chatbot and ask questions about it. ```

> ```python
> import gradio as gr
> 
> chat_input = gr.MultimodalTextbox(file_types=["image"], placeholder="Enter message or upload file...")
> ```
> 
> `gr.ChatInterface` also supports multimodality, simply pass in the `multimodal` parameter as `True`:

Reviewer: I would remove this part, since this guide is about `ChatInterface`, not the individual components.

```suggestion
```python
import gradio as gr

chat_input = gr.MultimodalTextbox(file_types=["image"], placeholder="Enter message or upload file...")
```

`gr.ChatInterface` also supports multimodality, simply pass in the `multimodal` parameter as `True`:
```

```python

> import gradio as gr

Reviewer: ```suggestion
import gradio as gr
import time

def echo(message, history):
    t = x['text']
    for i in range(len(t)):
        time.sleep(0.5)
        yield t[:i+1]

demo = gr.ChatInterface(fn=echo, examples=["hello", "hola", "merhaba"], title="Echo Bot", multimodal=True)

demo.launch()
```

> The `MultimodalTextbox` returns a dictionary of the form `{"text": "user input", "file": FileData}`. The FileData object contains the file name, file type, and file data. You can access the file path using the `file.path` attribute of the FileData object.

Reviewer: ```suggestion
When `multimodal=True`, the first parameter of your function should accept a dictionary that looks like this: `{"text": "user input", "file": ["file_path1", "file_path2", ...]}`.
```
